acoustid/data/source.py

```
@@ -22,32 +23,63 @@ def get_source(conn: AppDB, id: int) -> dict[str, Any] | None:
-        id = conn.execute(insert_stmt).inserted_primary_key[0]
-        logger.info(
-            "Inserted source %d with account %d and application %d (%s)",
-            id,
-            account_id,
-            application_id,
-            version,
+    return conn.execute(query).scalar()
+
+
+def find_or_insert_source(
+    conn: AppDB, application_id: int, account_id: int, version: str | None = None
```

Ruff (0.8.2)

Trailing comma missing

Add trailing comma

(COM812)